tests/MongoDB.Driver.Tests/Linq/Translators/AggregateProjectTranslatorTests.cs

```diff
@@ -1253,11 +1253,17 @@ public void Should_translate_split()
         {
             RequireServer.Check().VersionGreaterThanOrEqualTo("3.3.6");
 
+            /* for implementations that don't support omitted optional parameters in expression trees
+             * skip to next test */
+#if NETCOREAPP2_1 
+            var result = Project(x => new { Result = x.A.Split(new[] { 'e' }) }); 
+#else             
             var result = Project(x => new { Result = x.A.Split('e') });
             result.Projection.Should().Be("{ Result: { \"$split\": [\"$A\", \"e\" ] }, _id: 0 }");
             result.Value.Result.Should().BeEquivalentTo("Aw", "som", "");
 
             result = Project(x => new { Result = x.A.Split(new[] { 'e' }) });
+#endif
```

Comment:
I find this #if for lines 1258-1266 very confusing. Part of the problem is that the test reuses the same variable for 4 different scenarios, and the variable can only be declared using var (because of the anonymous type).

Consider using 4 variables instead and refactoring as follows:

    #if !NETCOREAPP2_1 
            /* for implementations that don't support omitted optional parameters in expression trees
             * skip to next test */
            var result1 = Project(x => new { Result = x.A.Split('e') });
            result1.Projection.Should().Be("{ Result: { \"$split\": [\"$A\", \"e\" ] }, _id: 0 }");
            result1.Value.Result.Should().BeEquivalentTo("Aw", "som", "");
    #endif

            var result2 = Project(x => new { Result = x.A.Split(new[] { 'e' }) });
            result2.Projection.Should().Be("{ Result: { \"$split\": [\"$A\", \"e\" ] }, _id: 0 }");
            result2.Value.Result.Should().BeEquivalentTo("Aw", "som", "");

            var result3 = Project(x => new { Result = x.A.Split(new[] { 'e' }, StringSplitOptions.None) });
            result3.Projection.Should().Be("{ Result: { \"$split\": [\"$A\", \"e\" ] }, _id: 0 }");
            result3.Value.Result.Should().BeEquivalentTo("Aw", "som", "");

            var result4 = Project(x => new { Result = x.A.Split(new[] { "es" }, StringSplitOptions.None) });
            result4.Projection.Should().Be("{ Result: { \"$split\": [\"$A\", \"es\" ] }, _id: 0 }");
            result4.Value.Result.Should().BeEquivalentTo("Aw", "ome");

In this version it is crystal clear that one test is being omitted when testing on NETCOREAPP2_1.

Reply:
I like this: I've taken your suggestion!